src/goto-symex/goto_symex_state.cpp

```diff
@@ -345,10 +347,12 @@ void goto_symex_statet::rename(
 
     INVARIANT(
       (expr.id() != ID_with ||
-       expr.type() == to_with_expr(expr).old().type()) &&
+       add_const(expr).type() == to_with_expr(add_const(expr)).old().type()) &&
```

Comment:
Maybe declare a `const exprt &c_expr = expr;` before the `INVARIANT` and use that in here? There are a lot of `add_const`s here.